src/esrs/postgres/mod.rs

```diff
@@ -251,62 +189,51 @@ impl<
+        Event: Serialize + DeserializeOwned + Send + Sync,
+        Error: From<sqlx::Error> + From<serde_json::Error> + Send + Sync,
+        Projector: PgProjector<Event, Error> + Send + Sync + ?Sized,
+        Policy: PgPolicy<Event, Error> + Send + Sync + ?Sized,
+    > ProjectorStore<Event, Transaction<'_, Postgres>, Error> for InnerPgStore<Event, Error, Projector, Policy>
 {
     fn project_event<'a>(
         &'a self,
-        store_event: &'a StoreEvent<Evt>,
-        executor: &'a mut PoolConnection<Postgres>,
-    ) -> Pin<Box<dyn Future<Output = Result<(), Err>> + Send + 'a>>
+        store_event: &'a StoreEvent<Event>,
+        transaction: &'a mut Transaction<Postgres>,
+    ) -> Pin<Box<dyn Future<Output = Result<(), Error>> + Send + 'a>>
     where
         Self: Sync + 'a,
     {
-        async fn run<
-            Ev: Serialize + DeserializeOwned + Send + Sync,
-            Er: From<sqlx::Error> + From<serde_json::Error> + Send + Sync,
-            Prj: PgProjector<Ev, Er> + Send + Sync + ?Sized,
-            Plc: PgPolicy<Ev, Er> + Send + Sync + ?Sized,
-        >(
-            me: &InnerPgStore<Ev, Er, Prj, Plc>,
-            store_event: &StoreEvent<Ev>,
-            executor: &mut PoolConnection<Postgres>,
-        ) -> Result<(), Er> {
-            for projector in &me.projectors {
-                projector.project(store_event, executor).await?
+        Box::pin(async move {
+            for projector in &self.projectors {
+                projector.project(store_event, transaction).await?
             }
 
             Ok(())
-        }
-
-        Box::pin(run::<Evt, Err, Projector, Policy>(self, store_event, executor))
+        })
     }
 }
 
 #[async_trait]
 impl<
-        Evt: Serialize + DeserializeOwned + Send + Sync,
-        Err: From<sqlx::Error> + From<serde_json::Error> + Send + Sync,
-        Projector: PgProjectorEraser<Evt, Err> + Send + Sync + ?Sized,
-        Policy: PgPolicy<Evt, Err> + Send + Sync + ?Sized,
-    > EraserStore<Evt, Err> for InnerPgStore<Evt, Err, Projector, Policy>
+        Event: Serialize + DeserializeOwned + Send + Sync,
+        Error: From<sqlx::Error> + From<serde_json::Error> + Send + Sync,
+        Projector: PgProjectorEraser<Event, Error> + Send + Sync + ?Sized,
+        Policy: PgPolicy<Event, Error> + Send + Sync + ?Sized,
+    > EraserStore<Event, Error> for InnerPgStore<Event, Error, Projector, Policy>
 {
-    async fn delete(&self, aggregate_id: Uuid) -> Result<(), Err> {
-        let mut connection: PoolConnection<Postgres> = self.begin().await?;
+    async fn delete(&self, aggregate_id: Uuid) -> Result<(), Error> {
+        let mut transaction: Transaction<Postgres> = self.pool.begin().await?;
         let _ = sqlx::query(self.queries.delete())
             .bind(aggregate_id)
-            .execute(&mut *connection)
+            .execute(&mut *transaction)
             .await
             .map(|_| ());
```

Comment:
We can bubble up the error here instead of ignoring it
```suggestion
            .execute(&mut *transaction)
            .await?;
```

Reply:
Yeah, you are right..